{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Notebook #7_ OMIM Colab_hgm.ipynb","provenance":[],"collapsed_sections":[],"authorship_tag":"ABX9TyOeaQG1oX4/GjnviYNsF2YN"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","metadata":{"id":"e4_-znEWZ8AB"},"source":["First you need to link your Google Drive to the notebook in order to access the files needed for this module.\n","\n","Run the cell below and follow instructions to mount the drive."]},{"cell_type":"code","metadata":{"id":"SHOuF0YNHktr"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"OrHpyyZhi0La"},"source":["## Installing Biopython\n","\n","At the beginning of each module, we will install **Biopython**. Biopython is a large open-source application programming interface (API) used in both bioinformatics software development and in everyday scripts for common bioinformatics tasks. It contains several packages that you will need to import which will allow you to run the analyses required for this project. \n","\n","REF:\n","* Cock, P. J., Antao, T., Chang, J. T., Chapman, B. A., Cox, C. J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F., Wilczynski, B., & de Hoon, M. J. (2009). Biopython: freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics (Oxford, England), 25(11), 1422â€“1423. https://doi.org/10.1093/bioinformatics/btp163\n"]},{"cell_type":"code","metadata":{"id":"7hOWkK_qiyt_"},"source":["!pip install biopython"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"yD8hKEUk3Ot4"},"source":["# Investigating the biological impact of the mutation and its possible role in human disease\n","For this section, your research will focus on investigating the biological impact of the mutation you are studying. To do this, you will use the OMIM and KEGG databases."]},{"cell_type":"markdown","metadata":{"id":"Pp2KQ7uEQJkm"},"source":["## OMIM Search for information on genetic diseases\n","\n","The **OMIM** (Online Mendelian Inheritance of Man) database contains short, referenced reviews about genetic loci and genetic diseases. It\n","can be a very useful resource for finding out what type of research has been done on a gene or a disease.\n","\n","REF:\n","* http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=OMIM"]},{"cell_type":"markdown","metadata":{"id":"LMm1YA03QaCu"},"source":["## Install and import the necessary packages:\n","\n","The **romim package** was created to query the OMIM database but it runs in R. \n","\n","**R** is another programming language so you will need to install **rpy2** to run R code in Google Colab.\n","\n","**Methods** and **remotes** are R packages that help us both install the package and use the functions in the code.\n","\n","The **XML** package will be used to read the results that you obtain from your database searches. \n","\n","REF:\n","* https://github.com/davetang/romim"]},{"cell_type":"code","metadata":{"id":"yX04cBc0tuEv"},"source":["# Install the rpy2 interface to run R code\n","%load_ext rpy2.ipython"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"cqq6keGK0ynj"},"source":["%%R # This must precede all R code in Colab, to allow R code to run \n","\n","# Installing the main package\n","# Note how different it is from Python code\n","remotes::install_github('davetang/romim')\n","\n","# Import the library associated with the package\n","library(romim)\n","\n","# Intalling several packages\n","install.packages('XML')\n","install.packages('methods')\n","install.packages(\"remotes\")\n","\n","# Press 1 and ENTER when prompted"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ilBRqKQGMlWH"},"source":["## Obtaining the ID number (called mim number) associated with the HMG gene entry in OMIM"]},{"cell_type":"code","metadata":{"id":"4hQQALHDGlVw"},"source":["%%R # This must precede all R code in Colab, to allow R code to run \n","\n","# To access OMIM, we will use this key which will work as our password to access the database\n","set_key('4PUvWRqSSD2BuprIVAP_VQ') \n","\n","# Write HMGCR in the parenthesis below to obtain the entry for the HMG gene\n","gene_to_omim('HMGCR', show_query=TRUE)\n"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Now, the link above is a file with our information. Let's download and parse it."],"metadata":{"id":"t04n8P9yl6ZB"}},{"cell_type":"code","source":["%%R\n","# Saving the file:\n","\n","# Paste the url between the quotes below\n","url <- '####'\n","\n","# Set destination (write the file path and a file name followed by .xml)\n","destfile <- '/content/drive/MyDrive/Colab_Notebooks/hmg_project_files/FILE NAME HERE.xml'\n","\n","# Download the file\n","download.file(url, destfile)\n"],"metadata":{"id":"UhastgNIkp7A"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["%%R\n","# Parse the file for readability\n","result <- xmlParse(file = 'WHERE IS THE FILE?')\n","\n","# Read the file\n","read <- read_xml('WHERE IS THE FILE?' )\n","\n","# Find the mim number\n","num <- xml_find_all(read, \".//mimNumber\")\n","\n","# Display the number\n","num"],"metadata":{"id":"sBWzAFDkqSrd"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"JO11cwOuNQzA"},"source":["Write down the mim number for the entry from the results above."]},{"cell_type":"markdown","metadata":{"id":"LCTL91M9NfUx"},"source":["Answer here"]},{"cell_type":"markdown","metadata":{"id":"D9YzHQGWberR"},"source":["## Using OMIM to obtain more information about the gene\n","This time you will search the OMIM 'Hgm' entry for information.\n","\n","The function 'get_omim' helps you do just that since we can set certain arguments to 'TRUE' and obtain specific information about the entry.\n","\n","Run the next cell to see a list of Arguments that you can access."]},{"cell_type":"code","metadata":{"id":"eib4aT-cqMRo"},"source":["%%R\n","help(get_omim)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# Obtain information from the entry"],"metadata":{"id":"baghE8ey219O"}},{"cell_type":"code","source":["# Search OMIM again to obtain description information\n","%%R\n","set_key('KEY HERE')\n","\n","# Using mim number to get the entry, set 'text' argument to true\n","omim_result <- get_omim(###, text = ###)\n","\n","# Save the xml to a file\n","saveXML(omim_result, file=\"FILE NAME HERE.xml\")"],"metadata":{"id":"9MzZcfme1cnB"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#@title Load the results by providing the file name in this form (include file extension .xml)\n","\n","# MAKING RESULTS LOOK GOOD\n","import xml.etree.ElementTree as ET\n","import csv\n","import pandas as pd\n","\n","file_name = \"\" #@param {type:\"string\"}\n","\n","tree = ET.parse(file_name)\n","root = tree.getroot()\n"," \n","Ref_data4 = open('refdata4.csv', 'w')\n"," \n","csvwriter = csv.writer(Ref_data4)\n","allele_head = []\n","\n"," \n","count = 0\n","for member in root.findall('.//textSection'):\n","    allele = []\n","    ref_list = []\n","\n","    if count == 0:\n","      des = member.find('.//textSectionTitle').tag\n","      allele_head.append(des)\n","      \n","      mut = member.find('.//textSectionContent').tag\n","      allele_head.append(mut)\n","     \n","      csvwriter.writerow(allele_head)\n","      count = count + 1\n","       \n","    des = member.find('.//textSectionTitle').text\n","    allele.append(des)\n","    \n","    mut = member.find('.//textSectionContent').text\n","    allele.append(mut)\n","\n","    csvwriter.writerow(allele)\n","  \n","Ref_data4.close()\n","\n","data4= pd.read_csv(\"refdata4.csv\")\n","pd.set_option('display.max_colwidth',10000)\n","\n","data4"],"metadata":{"id":"B9CNi81P1v-c","cellView":"form"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"J0XqkjVAeUa9"},"source":["## Answer the following questions:\n","Input your answer in the cell below each question and press SHIFT+ENTER."]},{"cell_type":"markdown","source":["Read the sections titled **Gene Function**, **Biochemical Features**"],"metadata":{"id":"Rjwo6ak_yUWN"}},{"cell_type":"markdown","metadata":{"id":"iTIm0QiSef5J"},"source":["1. What other role(s), if any, does HMG-CoA Reductase play besides cholesterol biosynthesis?\n","\n"]},{"cell_type":"markdown","metadata":{"id":"asWjhVSbekgV"},"source":["Answer here"]},{"cell_type":"markdown","source":["2. If an inhibitor was found that irreversibly inhibited HMG-CoA Reductase, so that no cholesterol could be formed, would this be desirable?  Why or why not?\n","\n"],"metadata":{"id":"PMkMlTJiyWn_"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"26EwstjbyaJT"}},{"cell_type":"markdown","source":["3. Describe how statin molecules bind HMG-CoA Reductase.  Are statins competitive or non-competitive inhibitors? (Refer to https://chem.libretexts.org/Courses/University_of_Arkansas_Little_Rock/CHEM_4320_5320%3A_Biochemistry_1/05%3A_Michaelis-Menten_Enzyme_Kinetics/5.4%3A_Enzyme_Inhibition for a review, if needed.)\n","\n"],"metadata":{"id":"BRYJ92vHybnB"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"MKvbP17nyf4F"}},{"cell_type":"markdown","source":["4. Before the crystal structure was solved of statins bound to HMG-CoA Reductase, researchers already knew what type of inhibitor statins were by performing enzyme assays in the presence of statins and plotting the data in a Lineweaver-Burke plot.  Based on your answer in question 4, which of the following plots describes what the researchers found?\n","\n"],"metadata":{"id":"hCeN5Xejyi1H"}},{"cell_type":"code","source":["from IPython.display import IFrame\n","IFrame('https://drive.google.com/file/d/1NRe4pv2vR3l4sVlhA7LbtcjkDStOQcnr/preview', width=600, height=300)"],"metadata":{"id":"U44dlKFy90U7"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"HNN7rsFMynwx"}},{"cell_type":"markdown","metadata":{"id":"SRgdDxXFNNX0"},"source":["## Using OMIM to obtain information about Hypercholesterolemia\n","This time you will search the OMIM 'HYPERCHOLESTEROLEMIA' entry for information.\n","\n","The function 'get_omim' helps you do just that since we can set certain arguments to 'TRUE' and obtain specific information about the entry.\n","\n","Run the next cell to see a list of Arguments that you can access."]},{"cell_type":"code","source":["# Search OMIM again to obtain description information\n","%%R\n","set_key('KEY HERE')\n","\n","# Using mim number to get the entry, set 'text' argument to true\n","omim_result <- get_omim(###, ### = ###)\n","\n","# Save the xml to a file \n","saveXML(omim_result, file='FILE NAME HERE.xml')"],"metadata":{"id":"imlvz_wPNNX0"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#@title Load the results by providing the file name in this form (include file extension .xml)\n","\n","# MAKING RESULTS LOOK GOOD\n","import xml.etree.ElementTree as ET\n","import csv\n","import pandas as pd\n","\n","file_name = \"\" #@param {type:\"string\"}\n","\n","tree = ET.parse(file_name)\n","root = tree.getroot()\n"," \n","Ref_data4 = open('refdata4.csv', 'w')\n"," \n","csvwriter = csv.writer(Ref_data4)\n","allele_head = []\n","\n"," \n","count = 0\n","for member in root.findall('.//textSection'):\n","    allele = []\n","    ref_list = []\n","\n","    if count == 0:\n","      des = member.find('.//textSectionTitle').tag\n","      allele_head.append(des)\n","      \n","      mut = member.find('.//textSectionContent').tag\n","      allele_head.append(mut)\n","     \n","      csvwriter.writerow(allele_head)\n","      count = count + 1\n","       \n","    des = member.find('.//textSectionTitle').text\n","    allele.append(des)\n","    \n","    mut = member.find('.//textSectionContent').text\n","    allele.append(mut)\n","\n","    csvwriter.writerow(allele)\n","  \n","Ref_data4.close()\n","\n","data4= pd.read_csv(\"refdata4.csv\")\n","pd.set_option('display.max_colwidth',10000)\n","\n","data4"],"metadata":{"id":"WguQhD87NNX0","cellView":"form"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Read the **Description**, **Pathogenesis**, **Clinical Management**, and **Gene Therapy** sections. "],"metadata":{"id":"RmIyLOyon6rr"}},{"cell_type":"markdown","source":["## Answer the following questions:##\n","Input your answer in the cell below each question and press SHIFT+ENTER."],"metadata":{"id":"MMJsQ5hobPl3"}},{"cell_type":"markdown","source":["5. What is the primary genetic defect in the disease?\n"],"metadata":{"id":"B5jaEPClbR4o"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"Na7_jstCbYV5"}},{"cell_type":"markdown","source":["6. How is HMG-CoA Reductase related to the disease, familial hypercholesterolemia?\n","\n"],"metadata":{"id":"iAWyF-Xbbhnt"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"HKOIHQP3bkq1"}},{"cell_type":"markdown","source":["7. Is your patient the only known case of someone with familial hypercholesterolemia not responding to statin therapy?  Explain your answer.\n"],"metadata":{"id":"WNY5NEzGbniS"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"nyf8UwD1bt6y"}},{"cell_type":"markdown","source":["8. What types of gene therapy have been tried? \n"],"metadata":{"id":"SDo7tGp-bvy0"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"VNOnr3RHbw5g"}},{"cell_type":"markdown","source":["9. Who are the authors for the article describing these experiments?\n"],"metadata":{"id":"aAKUPtD0oLnY"}},{"cell_type":"markdown","source":["Answer here"],"metadata":{"id":"hdeYsvGooNA5"}}]}